iosApp/iosApp/Pages/Search/Results/SearchResultsContainer.swift

```diff
@@ -39,113 +24,41 @@ struct SearchResultsContainer: View {
-        self.globalRepository = globalRepository
-        self.searchResultsRepository = searchResultsRepository
-        self.visitHistoryUsecase = visitHistoryUsecase
         self.didAppear = didAppear
         self.didChange = didChange
     }
 
-    func loadResults(query: String) {
-        Task {
-            resultsState = .loading
-            switch try await onEnum(of: searchResultsRepository.getSearchResults(query: query)) {
-            case let .ok(result):
-                let showRoutes = searchVM.routeResultsEnabled
-                let results = result.data
-                resultsState = if results.stops.isEmpty, !showRoutes || results.routes.isEmpty {
-                    .empty
-                } else {
-                    .results(results: results, includeRoutes: showRoutes)
-                }
-            case nil: resultsState = .error
-            case let .error(error):
-                resultsState = .error
-                debugPrint(error)
-            }
-        }
-    }
-
-    func loadVisitHistory() {
-        Task {
-            do {
-                latestVisits = try await visitHistoryUsecase.getLatestVisits()
-                    .compactMap { visit in
-                        switch onEnum(of: visit) {
-                        case let .stopVisit(stopVisit):
-                            // TODO: https://app.asana.com/0/1205425564113216/1208355161933391/f
-                            globalResponse?
-                                .stops[stopVisit.stopId]
-                                .map { stop in
-                                    StopResult(
-                                        id: stop.id,
-                                        rank: 0,
-                                        name: stop.name,
-                                        zone: nil,
-                                        isStation: true,
-                                        routes: []
-                                    )
-                                }
-                        }
-                    }
-            } catch {}
-        }
-    }
-
     func handleStopTap(stopId: String) {
-        guard let stop = globalResponse?.stops[stopId] else { return }
+        guard let stop = searchVM.getStopFor(id: stopId) else { return }
         nearbyVM.navigationStack.append(.stopDetails(stop, nil))
     }
 
-    func showRecentStops() {
-        resultsState = nil
-        // TODO: https://app.asana.com/0/1205425564113216/1208355161933391/f
-        // guard let latestVisits, !latestVisits.isEmpty else { return }
-        // resultsState = .recentStops(stops: latestVisits)
-    }
-
     var body: some View {
         SearchResultsView(
-            state: resultsState,
+            state: searchVM.resultsState,
             handleStopTap: handleStopTap
         )
+        .task {
+            await searchVM.getGlobalData()
+            await searchVM.loadSettings()
+            await searchVM.loadVisitHistory()
```

Comment:
nitpick: I'm not certain, but I think the way this is written, each of these are done in sequence, right? Could it speed things up slightly to have a separate task for each one?

Reply:
I think visitHistory at the very least benefits from being in this sequence but it does make sense to move loadSettings out!

Reply:
Cool, yeah, I wasn't sure which were dependent on each other. It might make sense to just add a helper function in searchVM to call all of them in sequence or parallel as needed, so you only need to do a single call here.